mdnsd/kiface.c

```diff
@@ -125,98 +66,17 @@ get_rtaddrs(int addrs, struct sockaddr *sa, struct sockaddr **rti_info)
 		if (addrs & (1 << i)) {
 			rti_info[i] = sa;
 			sa = (struct sockaddr *)((char *)(sa) +
-			    ROUNDUP(sa->sa_len, sizeof(long)));
+			    ROUNDUP(sa->sa_len));
 		} else
 			rti_info[i] = NULL;
 	}
 }
 
-int
-fetchifs(int ifindex)
-{
-	size_t			 len;
-	int			 mib[6];
-	char			*buf, *next, *lim;
-	struct if_msghdr	 ifm;
-	struct kif_node		*kif;
-	struct sockaddr		*sa, *rti_info[RTAX_MAX];
-	struct sockaddr_dl	*sdl;
-
-	mib[0] = CTL_NET;
-	mib[1] = AF_ROUTE;
-	mib[2] = 0;
-	mib[3] = AF_INET;
-	mib[4] = NET_RT_IFLIST;
-	mib[5] = ifindex;
-
-	if (sysctl(mib, 6, NULL, &len, NULL, 0) == -1) {
-		log_warn("sysctl");
-		return (-1);
-	}
-	if ((buf = malloc(len)) == NULL) {
-		log_warn("fetchif");
-		return (-1);
-	}
-	if (sysctl(mib, 6, buf, &len, NULL, 0) == -1) {
-		log_warn("sysctl");
-		free(buf);
-		return (-1);
-	}
-
-	lim = buf + len;
-	for (next = buf; next < lim; next += ifm.ifm_msglen) {
-		memcpy(&ifm, next, sizeof(ifm));
-		if (ifm.ifm_version != RTM_VERSION)
-			continue;
-		if (ifm.ifm_type != RTM_IFINFO)
-			continue;
-
-		sa = (struct sockaddr *)(next + sizeof(ifm));
-		get_rtaddrs(ifm.ifm_addrs, sa, rti_info);
-
-		if ((kif = calloc(1, sizeof(struct kif_node))) == NULL) {
-			log_warn("fetchifs");
-			free(buf);
-			return (-1);
-		}
-
-		kif->k.ifindex = ifm.ifm_index;
-		kif->k.flags = ifm.ifm_flags;
-		kif->k.link_state = ifm.ifm_data.ifi_link_state;
-		kif->k.media_type = ifm.ifm_data.ifi_type;
-		kif->k.baudrate = ifm.ifm_data.ifi_baudrate;
-		kif->k.mtu = ifm.ifm_data.ifi_mtu;
-		if ((sa = rti_info[RTAX_IFP]) != NULL)
-			if (sa->sa_family == AF_LINK) {
-				sdl = (struct sockaddr_dl *)sa;
-				if (sdl->sdl_nlen >= sizeof(kif->k.ifname))
-					memcpy(kif->k.ifname, sdl->sdl_data,
-					    sizeof(kif->k.ifname) - 1);
-				else if (sdl->sdl_nlen > 0)
-					memcpy(kif->k.ifname, sdl->sdl_data,
-					    sdl->sdl_nlen);
-				if (sdl->sdl_alen > 0) {
-					if (sdl->sdl_alen != ETHER_ADDR_LEN)
-						log_warnx("sdl->sdl_alen = %u\n",
-						    sdl->sdl_alen);
-					memcpy(&kif->k.ea,
-					    sdl->sdl_data + sdl->sdl_nlen,
-					    MIN(sizeof(kif->k.ea),
-					    sdl->sdl_alen));
-				}
-
-				/* string already terminated via calloc() */
-			}
-		kif_insert(kif);
-	}
-	free(buf);
-	return (0);
-}
-
 void
 kev_init(void)
 {
 	int		opt = 0, rcvbuf, default_rcvbuf;
+	int		rtfilter;
```

Comment:
If you're going to `#ifdef __OpenBSD__` the `rtfilter = ...` perhaps you could `#ifdef` the declaration for `rtfilter`.